### ISSUE TITLE
Unable To Start Windows Service

### ISSUE THREAD

**Issue by DanTheTechMan** [tgorgdotcom/locast2plex#121]
I can't seem to get the Windows service for Locast2Plex to run. I follow the guide to the letter. The only things I change are the path to Python and the Arguments for the service.  Here's what my screen looks like:

![image](https://user-images.githubusercontent.com/12401470/97379074-4d8c4280-189a-11eb-9f73-daa2e4c16aee.png)

The arguments I put for the service are as follows:
D:\Plex\Plex Media Server\Scripts\locast2plex\main.py -u:xxxxxxxxxx@xxxxx.xxx -p:xxxxxxxxxx --debug --addy:127.0.0.1 --port:6077

Note that I've replaced the all characters for the username and password with "x".

Here's the error message I get:

![image](https://user-images.githubusercontent.com/12401470/97379338-f2a71b00-189a-11eb-90ce-9d80f12c06f2.png)

Any help will be greatly appreciated.

**Comment by chris00780** [comment]
I'm seeing the same thing on windows 10. Not sure what I'm doing wrong either

**Comment by deathbybandaid** [comment]
The arguments were moved into the config file a while back. Please look at the config example for more Information

**Comment by DanTheTechMan** [comment]
Does that mean I can no longer create a Windows service and avoid having to open a Command Line window after every reboot?

**Comment by deathbybandaid** [comment]
Not at all. Put those settings in the config file instead of cmdline args. Not sure about nssm, but I know running things via cmd, you may need quotes around the paths that have spaces (ie "Program Files")

**Comment by DanTheTechMan** [comment]
I have it running successfully via the command line. I'm using it as I type this. However, I'm trying to have a more automated solution. I would like to be able to, as the tutorial suggests, automatically load on reboot, just restart the service when updates are made to the py scripts, and not need to have a windows command line window open. Is that still possible?

**Comment by deathbybandaid** [comment]
It should be. I haven't touched nssm in a very long time

**Comment by deathbybandaid** [comment]
Try setting the nssm startup directory to the l2p folder

**Comment by tharris9d** [comment]
It works great for me. Make sure you have the python 3 version, fill in the config.ini and just point to main.py in nssm.

**Comment by DanTheTechMan** [comment]
I have Python3 already, and I already I filled in the config.ini file. It runs fine via the command line, however I don't want to have to go through the process of starting it every time I reboot.

I am now getting a different error:

![image](https://user-images.githubusercontent.com/12401470/97504175-baf8ab80-194c-11eb-8f5f-2930919c609a.png)

I looked at the System Event Viewer for details on this error, and the error is occurring because the system can't find one of the files or folders, however that's all it says. It doesn't tell me which file or folder can't be located.

Also, do I need to add all the additional arguments still, or do I just point to Python in the "Path" field, and the main.py script in the "Arguments" field, without any additional arguments (such as Username, Password, IP:Port)?

**Comment by tharris9d** [comment]
I just installed a new service successfully. Make sure you point to main.py without ANY arguments. All options are in the config.
Use srvman.exe to make it easy to delete any unwanted services. If you dont have it check my wiki article. i have the website listed there. i need to update it to remove the command line args.

**Comment by DanTheTechMan** [comment]
Ok. Thanks. This is what my NSSM window looks like:

![image](https://user-images.githubusercontent.com/12401470/97506327-1167e900-1951-11eb-8cc7-75e51ace4247.png)

Path: "C:\Program Files\Python38\python.exe"
Startup Directory: D:\Plex\Plex Media Server\Scripts\locast2plex
Arguments: "D:\Plex\Plex Media Server\Scripts\locast2plex\main.py"
Service Name: Locast2Plex

Is that correct?

**Comment by tharris9d** [comment]
Yes! Make sure you don't already have a service named Locast2Plex

**Comment by DanTheTechMan** [comment]
Thanks, however, it doesn't work for me. I get this error again:

![image](https://user-images.githubusercontent.com/12401470/97504175-baf8ab80-194c-11eb-8f5f-2930919c609a.png)

Any ideas as to what the issue could be?

**Comment by DanTheTechMan** [comment]
Does anyone have any other ideas on how to remedy this issue?

**Comment by chris00780** [comment]
> Does anyone have any other ideas on how to remedy this issue?

Upgrading Python did it for me. Point nssm to the new python and the arguments with no quotes to main.py. 

Make sure you have the config.ini filled out with your username and password. Mine is working perfectly

**Comment by DanTheTechMan** [comment]
Unfortunately, upgrading Python isn't an option for me at the moment, as I'm on the most recent version of Python that supports my OS. The next upgrade of Python compatible with my OS won't be released until Mid-November, so, unless anyone has any other ideas that don't require me waiting for something, it looks like I'm SOL for the moment. Also, as I've said multiple times, I already have it working via the command line (which means the config.ini file is already filled out), however, what I'm ultimately looking for is a completely automated solution (for the script to load automatically with my OS (and Plex), without any intervention from me), which, if I can get this set up correctly and working, will be exactly what I am looking for. The only time I should have to intervene is when the script(s) get updated, and even then, it should be as simple as 1-2-3 (stopping the service, overwriting the files, and starting the service back up). Also, removing the quotes returns me to the original error.

**Comment by deathbybandaid** [comment]
I'd say, you need to research on google how to use nssm for a python script.

**Comment by tharris9d** [comment]
@DanTheTechMan Please put your command line in a batch/cmd file and run it to make sure it works.

Here is mine. Please test it on your system and confirm it's running.
In a locast2plex.cmd file place your path to python and main.py
   echo off
   cls
   "C:\Users\Valued Customer\AppData\Local\Programs\Python\Python38-32\python.exe" C:\code\locast2plex\main.py
   rem pause

nssm.exe
In nssm.exe put in your path to Python 
in arguments put C:\code\locast2plex\main.py (or whatever your path is to main.py)

**Comment by tharris9d** [comment]
Here is what my successfull NSSM ui looks like.
![image](https://user-images.githubusercontent.com/217445/97744432-72ec9c80-1aa4-11eb-9b41-c9c0f395eeaf.png)

**Comment by DanTheTechMan** [comment]
I tried it exactly as you said and I still get an error. I have no idea why it's not working.

**Comment by tharris9d** [comment]
So you can run it from the cmd file successfully?

**Comment by DanTheTechMan** [comment]
Yes.

**Comment by tharris9d** [comment]
Try moving your code out of the plex scripts folder to c:\code\locast2Plex. Just to see....

**Comment by tharris9d** [comment]
Check services to make sure your non working services are deleted.

**Comment by DanTheTechMan** [comment]
Ok. I'll try that. I delete the service after every failed test, before I create it again, so there should be nothing to delete.

**Comment by DanTheTechMan** [comment]
Well, I finally came to a working solution. Here are the steps I took to get fully operational: As a test, as @tharris9d  recommended, I copied the "locast2plex" folder to the "C" drive, then edited the arguments of the "Locast2Plex" service with NSSM (as I apparently forgot to delete it after my last failed test) to point to the new temporary location of the "locast2plex" folder. That worked, however, I did not want to run it from that location permanently, so I created a Directory Junction to point to the original location of the "locast2plex" folder (on the "D" drive), to see if I could fool the service to thinking it was still being run from the "C" drive. That worked like a charm. I am now fully up and running, exactly as I wanted to be. Thank you to everyone for all your help and advice, and a BIG thanks to @tharris9d for your diligence and for helping me get to this point.

**Comment by tharris9d** [comment]
Awesome! I am so happy you got it working.

**Comment by kogren161** [comment]
I got the service up and running, but both emby and plex give me errors trying to find the HDhomerun device.

This is my config.ini
[main]
locast_username = <xxxxxx@xxxxx.com>
locast_password = <xxxxxxx>
plex_accessible_ip = 127.0.0.1:6077
c = \locast2plex\main.py
uuid = mlliqmoq

The last line was tacked on by some other process.

I have tried it with and without the port info (not sure if the context is correct)

**Comment by tgorgdotcom** [comment]
Hello @kogren161.   

Noticed a few items with the ini file:
 - `c = \locast2plex\main.py` can be removed
 - The `:6077` part of the plex_accessible_ip can be removed

Try seeing if making the changes to the ini resolves your issue.

**Comment by kogren161** [comment]
Thank you for the suggestions, I was able to get this running in windows terminal, as well as successfully setting up the service and starting it.  I was able to find the tuner in Plex as well as Emby.  When I try to tune to a channel in Plex I get this error 

"Playback Error
Could not tune channel. Please check your tuner or antenna."  

I have not setup a premium account, and don't appear to be able to go any further in Emby without a premium account.

**Comment by kogren161** [comment]
I thought perhaps that I needed to "install" ffmpeg, so I went through the instructions on youtube to do that and rebooted.  Unfortunately now Plex can't see the tuner.

**Comment by kogren161** [comment]
I shut down the service, and restarted via command line.  Plex found the tuner, and now will tune to the channels.

**Comment by kogren161** [comment]
Although the service appears to be running, Locast2Plex only seems to be working from the command line.

**Comment by kogren161** [comment]
I found that I have the path to the Python shortcut rather than the path to the file.  It included a space.

**Comment by kogren161** [comment]
No dice :( still only runs from the command line.

**Comment by tharris9d** [comment]
Awesome. Whenever you update the code to a newer version simple restart the service! Have fun.
